datalad_deprecated/ls_webui.py

```diff
@@ -347,6 +355,18 @@ def ds_traverse(rootds, parent=None, json=None,
     fsparent = fs_extract(parent.path, parent.repo, basepath=rootds.path) \
         if parent else None
 
+    # from my understanding, this constructs a file metadata map that maps file path to external urls found 
+    # (the map is passed to fs_traverse)
+    metadata_raw = dl.extract_metadata(types=["datalad_core"], dataset=rootds)
+    metadata_map = {}
+    
+    for file_metadata in metadata_raw:
+        if file_metadata['type'] == 'file':
+            try:
+                metadata_map[file_metadata['path']] = file_metadata['metadata']['datalad_core']['url']
+            except ValueError:
```

Comment:
better make things explicit (checks) instead of just relying on silently swallowing ValueError.  Since it is not even `KeyError` -- why/when ValueError happens?

Reply:
I think ValueError sometimes occurs when there is no URL found from datalad Core, but I think my check on line 364 seems to take care of this, after testing it out a bit. I think the try is redundant now, but I'll add a bit to the if statement just in case.